pytorch_lightning/trainer/data_loading.py

```diff
@@ -308,20 +290,16 @@ def _reset_eval_dataloader(
                 # percent or num_steps
                 limit_eval_batches = getattr(self, f'limit_{mode}_batches')
 
-                if num_batches != float('inf'):
-                    self._check_batch_limits(f'limit_{mode}_batches')
-
-                    # limit num batches either as a percent or num steps
-                    if isinstance(limit_eval_batches, float):
-                        num_batches = int(num_batches * limit_eval_batches)
-                    else:
-                        num_batches = min(len(dataloader), limit_eval_batches)
-
-                elif limit_eval_batches not in (0.0, 1.0):
+                # limit num batches either as a percent or num steps
+                if isinstance(limit_eval_batches, int) or limit_eval_batches == 0.0:
```

Comment:
We could probably benefit from refactoring this into a function, since this similar to the code that handles `num_training_batches`.

Reply:
@SkafteNicki mind do it as it roots repo branch, you are free to commit your suggestion... :]
just make it as wip if you decide to do any edits 🐰

Reply:
@Borda I will just follow up with a small PR and approve this